Aligning use of counter to the upstream changes (Kconfig->DT transition)

nordic-krch opened the pull request.

nordic-krch committed:
tests: zigbee: osif: timer_counter: Align to zephyr changes

Align test to changes in the counter driver (transition to DT
configuration.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: zigbee: shell: Add overlays enabling counter

Counter is now controlled by the device tree and it must be
enabled in the sample.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
zigbee: osif: Remove use of CONFIG_COUNTER_TIMER2

Kconfig option was removed and it must be replaced with
device tree node status check.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: zigbee: light_switch: Add overlays enabling counter

Counter is now controlled by the device tree and it must be
enabled in the sample.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: zigbee: light_bulb: Add overlays enabling counter

Counter is now controlled by the device tree and it must be
enabled in the sample.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: zigbee: template: Add overlays enabling counter

Counter is now controlled by the device tree and it must be
enabled in the sample.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: drivers: lpuart: Align to counter changes

Align to change in counter configuration (from Kconfig to
devicetree).

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: peripheral: lpuart: Adapt to counter changes

Adapt sample to changes in counter driver configuration which is
now done through the devicetree.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
gazell: Adapt to changes in counter

Align to changes in counter driver configuration (from Kconfig
to devicetree).

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: zigbee: osif: nvram: Align to zephyr changes

Align to configuration counter change (Kconfig to devicetree).

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
boards: arm: Cleanup after counter driver update

After moving counter driver configuration to the devicetree timer
nodes shall not be enabled if not used. Cleaning up board dts
files.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
applications: zigbee_weather_station: Align to zephyr changes

Align application to change in counter driver configuration.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: zigbee: zboss_api: alarm_api: Align to zephyr changes

Align test to change in counter driver configuration.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: zigbee: zboss_api: callback_api: Align to zephyr changes

Align test to change in counter driver configuration.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: zigbee: network_coorindator: Align to zephyr changes

Align sample to change in counter driver configuration.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

mbolivar-nordic commented on applications/zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/app.overlay:
```diff
@@ -14,6 +14,10 @@
 	};
 };
 
+zigbee_timer: &timer2 {
```
I do not think a magic node label is the right approach. I believe that either a property of the radio node (`zigbee-timer`, similar to the `fem` property in https://docs.zephyrproject.org/latest/build/dts/api/bindings/net/wireless/nordic%2Cnrf-radio.html#dtbinding-nordic-nrf-radio) or a new NCS specific chosen node (`ncs,zigbee-timer`) is a better design.

I know we have some magic node labels in NCS (`nrf_radio_fem`), but when discussing this idea with the downstream author responsible for this, I said explicitly this was not an upstreamable solution, and now I regret even OKing this for NCS and I think it would be a bad idea to introduce more cases.

Node labels are convenient but they cannot be easily overridden. You would have to delete the entire node, restore its state exactly (which is not robust or future-proof if done from an overlay, since there could be other overlays changing its properties that you don't know about) but without the node label, and then assign the node label to another node.

By contrast, overriding a property (either with `&radio { prop = ... };` or by overriding a `/chosen` property) is easy, and it doesn't have the future-proofing problems described above.

Can you please pick one of the two options and rework this accordingly?

nordic-krch replied:
Using chosen would look like that? https://github.com/mbolivar-nordic/sdk-nrf/pull/22/commits/69734b7e4a201757b3c7d2df3bd3294414bb28e7 Is that what you had on your mind? If that looks ok i will apply changes to other samples.

Regarding adding property to the radio node. That would mean adding it to zephyr? Probably it would need to stay in downstream only since upstream has no zigbee support.

mbolivar-nordic replied:
> Using chosen would look like that? https://github.com/mbolivar-nordic/sdk-nrf/commit/69734b7e4a201757b3c7d2df3bd3294414bb28e7 Is that what you had on your mind?

Yep, that looks great, thanks.

> If that looks ok i will apply changes to other samples.

OK! Thank you.

> Regarding adding property to the radio node. That would mean adding it to zephyr? Probably it would need to stay in downstream only since upstream has no zigbee support.

It could be an `[nrf noup]` to our binding for the nordic,nrf-radio node if you wanted to do it as a property. But either way is fine with me. I trust your judgement on which one will work best for NCS.

lemrey commented:
There are some failures in the `modem_trace_backend` sample which seem related to this changes as well.
It has a dependency on the timer peripheral through `cpu_load`.

nordic-krch committed:
debug: cpu_load: Align timer configuration to upstream changes

Counter configuration is moved fully to DT and HAS_HW_NRF_TIMERx
cannot longer be used to determine if given TIMER instance is
present. Switching to check if not has timer compat.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch commented:
> There are some failures in the modem_trace_backend sample which seem related to this changes as well.
It has a dependency on the timer peripheral through cpu_load.

Fixed.

mbolivar-nordic commented on subsys/zigbee/Kconfig:
```diff
@@ -362,13 +362,16 @@ config ZIGBEE_TC_REJOIN_ENABLED
 	prompt "ZBOSS time source"
-	default ZIGBEE_TIME_COUNTER if ZIGBEE_LIBRARY_PRODUCTION
+	default ZIGBEE_TIME_COUNTER if ZIGBEE_LIBRARY_PRODUCTION && \
+		$(dt_chosen_enabled, ncs_zigbee_timer)
```
This doesn't look right; did you test this?

nordic-krch replied:
It seems to work, if i comment `chosen` then default changes. What would be the issue?

mbolivar-nordic replied:
You're giving the function the argument `" ncs_zigbee_timer"` (note leading space and use of underscores). I expect argument `"ncs,zigbee-timer"`

mbolivar-nordic commented on samples/zigbee/network_coordinator/boards/nrf5340dk_nrf5340_cpuapp.overlay:
```diff
@@ -0,0 +1,16 @@
+
+ / {
+	chosen {
+		nordic,pm-ext-flash = &mx25r64;
```
Is this necessary?

nordic-krch replied:
removed
